[BUG]: nvcc fatal : Unsupported gpu architecture 'compute_' when running cmake -S . -B build

## Description

I ran into an error (`nvcc fatal : Unsupported gpu architecture 'compute_'`) running `cmake` in `icicle/`.

### Reproduce

I entered the `icicle/` directory and ran `cmake -S . -B build`. The output was:

```
CMake Error at /usr/share/cmake-3.22/Modules/CMakeDetermineCompilerId.cmake:726 (message):
  Compiling the CUDA compiler identification source file
  "CMakeCUDACompilerId.cu" failed.

  Compiler: /usr/local/cuda-12.2/bin/nvcc

  Build flags:

  Id flags: --keep;--keep-dir;tmp;-gencode=arch=compute_,code=sm_ -v

  

  The output was:

  1

  nvcc fatal : Unsupported gpu architecture 'compute_'

  

  

Call Stack (most recent call first):
  /usr/share/cmake-3.22/Modules/CMakeDetermineCompilerId.cmake:6 (CMAKE_DETERMINE_COMPILER_ID_BUILD)
  /usr/share/cmake-3.22/Modules/CMakeDetermineCompilerId.cmake:48 (__determine_compiler_id_test)
  /usr/share/cmake-3.22/Modules/CMakeDetermineCUDACompiler.cmake:298 (CMAKE_DETERMINE_COMPILER_ID)
  CMakeLists.txt:13 (project)


CMake Error: CMAKE_CXX_COMPILER not set, after EnableLanguage
-- Configuring incomplete, errors occurred!
See also "/home/user/icicle/icicle/build/CMakeFiles/CMakeOutput.log".
See also "/home/user/icicle/icicle/build/CMakeFiles/CMakeError.log".
```

## Expected Behavior

`cmake` should succeed.

## Environment

OS + Version: Ubuntu 22.04.3 LTS, Linux 5.15.0-79-generic x86_64

Cargo Version: `cargo 1.68.2 (6feb7c9cf 2023-03-26)`

GPU type: NVIDIA Corporation GP108GLM [Quadro P520] (rev a1)

## Additional context

`nvcc --version`:

```
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2023 NVIDIA Corporation
Built on Tue_Jul_11_02:20:44_PDT_2023
Cuda compilation tools, release 12.2, V12.2.128
Build cuda_12.2.r12.2/compiler.33053471_0
```

--- a/README.md
+++ b/README.md
@@ -28,7 +28,13 @@ ICICLE is a CUDA implementation of general functions widely used in ZKP. ICICLE 
 
 ## Build and usage
 
-> NOTE: [NVCC] is a prerequisite for building.
+
+### Prerequisites
+
+- [NVCC]
+- cmake 3.24.0 and above (though 3.22.1 has been tested and works as well)
+
+### Steps
 
 1. Define or select a curve for your application; we've provided a [template][CRV_TEMPLATE] for defining a curve
 2. Include the curve in [`curve_config.cuh`][CRV_CONFIG]

--- a/README.md
+++ b/README.md
@@ -32,7 +32,7 @@ ICICLE is a CUDA implementation of general functions widely used in ZKP. ICICLE 
 ### Prerequisites
 
 - [NVCC]
-- cmake 3.24.0 and above (though 3.22.1 has been tested and works as well)
+- cmake 3.18 and above
 
 ### Steps
 

--- a/icicle/CMakeLists.txt
+++ b/icicle/CMakeLists.txt
@@ -7,7 +7,9 @@ set(CMAKE_CUDA_STANDARD_REQUIRED TRUE)
 set(CMAKE_CXX_STANDARD_REQUIRED TRUE)
 # add the target cuda architectures
 # each additional architecture increases the compilation time and output file size
-if (NOT DEFINED CMAKE_CUDA_ARCHITECTURES)
+if (${CMAKE_VERSION} VERSION_LESS "3.24.0")
+    set(CMAKE_CUDA_ARCHITECTURES ${CUDA_ARCH})
+else()
     set(CMAKE_CUDA_ARCHITECTURES native) # on 3.24+, on earlier it is ignored, and the target is not passed
 endif ()
 project(icicle LANGUAGES CUDA CXX)

